### PR TITLE
Added requirement in new mod dialog that internal name begins with a capital

### LINKED ISSUE #2831: Require in new mod dialog that internal name begins with a capital
to make sure the user made mods have more consistent names

### PR DESCRIPTION
**Pull request by Generatoror**
**Brief Description of What This PR Does**

Added requirement in new mod dialog that internal name begins with a capital

**Related Issues**
closes #2831

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Otherwise looks fine but I'd just like to a little bit of extra clarity in the error message.

**Comment by hhyyrylainen** [comment]
When linking the issue that a PR closes, you need to use the "closes" keyword for that to be detected, I've edited the description to fix that.

File: locale/en.po [hunk 1]
```diff
@@ -3718,14 +3718,18 @@ msgid "INTERNAL_NAME_REQUIRED"
 msgstr "Internal name is required"
 
 #: ../src/modding/NewModGUI.cs:244
+msgid "INTERNAL_NAME_REQUIRES_CAPITAL"
+msgstr "Internal name should begin with a capital"
```
[the remark this answers is not in the document]
Comment (Generatoror) [reply locale/en.po]: I agree, thanks

**Review by hhyyrylainen — APPROVED** [review]
Looks fine to me, just needs to resolve the merge conflicts before this can be merged.

**Comment by Generatoror** [comment]
Ok, any particular thing i should take into account when resolving these merge conflicts?

**Comment by hhyyrylainen** [comment]
You can accept just most stuff from master, just careful to not ovewrite your English translations. After that you can re-run the update translation script to get other languages and comments in the .po files adjusted.

**Comment by RacerBG** [comment]
Everything seems to work fine. In order to spice things up I used cyrillic characters as well - no problems at all.

File: locale/en.po (hunk 1 above)
[the remark this answers is not in the document]
Comment (hhyyrylainen) [reply locale/en.po:3722]: Looks like this got reverted in the merge...

[the remark this answers is not in the document]
Comment (Generatoror) [reply locale/en.po:3722]: Weird, i thought i had double-checked, anyways, should cause no problems to commit it again

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**